client/network_create.go

```diff
@@ -13,6 +14,10 @@ func (cli *Client) NetworkCreate(ctx context.Context, name string, options types
 		NetworkCreate: options,
 		Name:          name,
 	}
+	if versions.LessThan(cli.version, "1.44") {
```

Comment:
Note to self; I'm actually wondering now if these checks we have in the client _work_, as we moved negotiating the API version to be as late as possible, which (I _think_) in this case _could_ mean that we postpone it to `cli.post` (which calls `cli.sendRequest`, which calls `cli.getAPIPath` (which handles the negotiation)) 🙈

Reply:
(It's probably fine to keep this as-is for now, as it looks like it's something to fix for other parts as well)

Reply:
opened a PR to fix this issue;

- https://github.com/moby/moby/pull/46463